src/net_processing.cpp

```diff
@@ -3497,9 +3492,11 @@ bool PeerLogicValidation::SendMessages(CNode* pto)
 
                     // If the peer's chain has this block, don't inv it back.
                     if (!PeerHasHeader(&state, pindex)) {
-                        pto->PushInventory(CInv(MSG_BLOCK, hashToAnnounce));
-                        LogPrint(BCLog::NET, "%s: sending inv peer=%d hash=%s\n", __func__,
-                            pto->GetId(), hashToAnnounce.ToString());
+                        CInv inv(MSG_BLOCK, hashToAnnounce);
+                        if (pto->PushInventory(inv)) {
+                            LogPrint(BCLog::NET, "%s: sending inv %s peer=%d\n",
```

Comment:
It seems we can move this logging inside ```PushInventory``` for cases where inv is for block?
Then this code won't be duplicate in 2 places.

I'm also wondering whether we should put *all* logging (including transactions) there to have a uniform behavior, not sure though.

Reply:
The main reason I didn't do that was to avoid multiple log entries when relaying to all peers, as in `RelayInventory`. A single message in that case is a lot less noisy.